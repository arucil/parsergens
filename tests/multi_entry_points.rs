
use insta::assert_debug_snapshot;

parsergens::parsergen!(mod prog_lalr_parser: LR("fixtures/prog.pg"));

#[test]
fn lalr() {
  let result = prog_lalr_parser::with_input(r"foo(1+2, 45*-2); bar(); baz(--((2)),1)")
    .parse_prog();
  assert_debug_snapshot!(result);

  let result = prog_lalr_parser::with_input(r"45*-2")
    .parse_expr();
  assert_debug_snapshot!(result);

  let result = prog_lalr_parser::with_input(r"foo((-2),45*-2)")
    .parse_proc_call();
  assert_debug_snapshot!(result);
}
