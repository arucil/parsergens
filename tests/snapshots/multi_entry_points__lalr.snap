---
source: tests/multi_entry_points.rs
expression: result
---
Ok(
    [
        ProcCall(
            "foo",
            [
                Binary(
                    Add,
                    Number(
                        1,
                    ),
                    Number(
                        2,
                    ),
                ),
                Binary(
                    Mul,
                    Number(
                        45,
                    ),
                    Neg(
                        Number(
                            2,
                        ),
                    ),
                ),
            ],
        ),
        ProcCall(
            "bar",
            [],
        ),
        ProcCall(
            "baz",
            [
                Neg(
                    Neg(
                        Number(
                            2,
                        ),
                    ),
                ),
                Number(
                    1,
                ),
            ],
        ),
    ],
)
